lib/model/QubitDigitalObject.php

```diff
@@ -1929,39 +1935,57 @@ public function createRepresentations($usageId, $connection = null)
     /**
      * Set 'page_count' property for this asset.
      *
-     * NOTE: requires the ImageMagick library
+     * Prefers to use the Imagick extension if it's loaded. Otherwise, uses pdinfo or ImageMagick
      *
      * @param null|mixed $connection
      *
      * @return QubitDigitalObject this object
      */
     public function setPageCount($connection = null)
     {
-        if ($this->canThumbnail() && sfImageMagickAdapter::isImageMagickAvailable()) {
-            $filename = ($this->derivativesGeneratedFromExternalMaster($this->usageId)) ? $this->getLocalPath() : $this->getAbsolutePath();
-
-            $extension = pathinfo($filename, PATHINFO_EXTENSION);
+        $filename = ($this->derivativesGeneratedFromExternalMaster($this->usageId)) ? $this->getLocalPath() : $this->getAbsolutePath();
+        $extension = pathinfo($filename, PATHINFO_EXTENSION);
+        $pages = null;
+
+        if ($this->canThumbnail()) {
+            if (extension_loaded('imagick')) {
+                // Only read metadata for performance
+                $im = new Imagick();
+                $im->pingImage($filename);
+                $pages = $im->count();
+                $im->clear();
+            } elseif ('pdf' == strtolower($extension) && sfImageMagickAdapter::pdfinfoToolAvailable()) {
+                trigger_error(
+                    'Using pdfinfo shell commands is deprecated in setPageCount. Install the Imagick PHP extension for better performance and security.',
+                    E_USER_DEPRECATED,
+                );
 
-            // If processing a PDF, attempt to use pdfinfo as it's faster
-            if ('pdf' == strtolower($extension) && sfImageMagickAdapter::pdfinfoToolAvailable()) {
                 $pages = sfImageMagickAdapter::getPdfinfoPageCount($filename);
-            } else {
-                $command = 'identify '.$filename;
+            } elseif (sfImageMagickAdapter::isImageMagickAvailable()) {
```

Comment:
I've been thinking about this logic to fall back to using ImageMagick in the event that the Imagick PHP extension is not present. I think that while this may provide some convenience with respect to AtoM upgrade requirements, I think it is more important to completely remove the original way of calling ImageMagick with exec() calls and rely solely on your new work to introduce Imagick. Removing the calls to exec() entirely will avoid triggering alerts in future security scans and remove the risk associated with these calls. In this case, if we remove all dependency on sfImageMagickAdapter.class.php, then we might be able to delete sfImageMagickAdapter.class.php entirely (I have not checked if there are other locations where it is referenced).